htp/htp_decompressors.c

```diff
@@ -207,6 +207,8 @@ static htp_status_t htp_gzip_decompressor_decompress(htp_decompressor_gzip_t *dr
         dout.tx = d->tx;
         dout.data = NULL;
         dout.len = 0;
+        // We should be called for a connection end and do
```

Comment:
I don't understand this comment

Reply:
This is indeed complicated.
So I will try to explain in a longer way.

The bigger question is do we have to output all the results we can with the data we get ? Or wait for more data to give a better result...
In this case, do we have to stream/output all the decompressed data (even if we did not reach content-length yet) as soon as we have it ? 
Or should we bufferize it and output it only when it is complete ?
In the latter case, `htp_gzip_decompressor_decompress` needs to be called in the case of a connection end so that it can still output the partially decompressed data.
But there is now no such mechanism in suricata/libhtp as to stop the connection but process the remaining buffered data.

This question comes again with other cases where we probe further data in order to interpret the current data, for instance https://github.com/OISF/libhtp/pull/195#issuecomment-478913730

Is this clearer Victor ?